numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -619,6 +620,10 @@ typedef struct _PyArray_Descr {
          * for NumPy 1.7.0.
          */
         NpyAuxData *c_metadata;
+        /* Cached hash value (-1 if not yet computed).
+         * This was added for NumPy 2.0.0.
+         */
+        npy_hash_t hash;
```

Comment:
Would the general scheme work like PyArray_Fields currently does? Which dtype fields should be private vs. public exactly?